addons/l10n_it_edi/wizard/account_move_send.py

```diff
@@ -1,12 +1,216 @@
+        for move, prepared_data in prepared_data_list.items():
+            if move.l10n_it_edi_state != 'being_sent':
+                continue
+
+            content = b64encode(prepared_data['l10n_it_edi_content'].encode()).decode()
+            filename = prepared_data['l10n_it_edi_filename']
+            if not move.commercial_partner_id._is_pa():
+                files_to_upload.append({'filename': filename, 'xml': content})
+            else:
+                pa_files.append({'move': move, 'filename': filename, 'xml': content})
+
+            attachments[move.id] = filename
+            moves[filename] = move
+
+        # Upload the files
+        if files_to_upload:
+            try:
+                results = self._get_proxy_user()._l10n_it_edi_upload(files_to_upload)
+            except AccountEdiProxyError as e:
+                for _move_id, filename in attachments.items():
+                    move.sudo().message_post(body=Markup(_("Error uploading the e-invoice file %s.<br>%s", filename, e.message)))
+                raise UserError(e.message)
+
+            # Handle results
+            for filename, vals in results.items():
+                move = moves[filename]
+                if 'error' in vals and vals.get('error_level', '') == 'error':
+                    move.l10n_it_edi_state = False
+                    move.l10n_it_edi_transaction = False
+                    move.sudo().message_post(body=Markup(_("Error uploading the e-invoice file %s.<br>%s", filename, vals['error'])))
+                else:
+                    move.l10n_it_edi_state = 'processing'
+                    id_transaction = vals['id_transaction']
+                    move.l10n_it_edi_transaction = id_transaction
+                    if id_transaction == 'demo':
+                        move.sudo().message_post(body=_("We are simulating the sending of the e-invoice file %s, as we are in demo mode.", filename))
+                    else:
+                        move.sudo().message_post(body=_("The e-invoice file %s was sent to the SdI for processing.", filename))
+
+        # Handle PA files
+        for vals in pa_files:
+            move = vals['move']
+            move.l10n_it_edi_state = 'require_user_signature'
+            move.l10n_it_edi_attachment_file = False
+            move.l10n_it_edi_transaction = False
+            move.sudo().message_post(body=Markup(_(
+                "Sending invoices to Public Administration partners is not supported.<br>"
+                "The IT EDI XML file is generated, please sign the document and upload it "
+                "through the 'Fatture e Corrispettivi' portal of the Tax Agency."
+            )))
+
+    def _link_invoice_documents(self, invoice, invoice_data):
+        # EXTENDS 'account'
+        super()._link_invoice_documents(invoice, invoice_data)
+        if (content := invoice_data.get('l10n_it_edi_content')):
+            self.env['account_edi_proxy_client.user']._l10n_it_edi_store(
```

Comment:
The attachment is something stored on the invoice. I don't understand why you put of that stuff on the proxy user. It doesn't make any sense to me.

Reply:
You think of the existing process and try to apply the same logic to Italian EDI. You see the only the export side and not the import one.  In the import, we have the attachment before we have the invoice, so we create it and commit it before even trying to parse it, so the method is placed on the frontier between Odoo and the EDI: the Proxy user.
I'll be duplicating the code, since you seemingly want to bind and couple every single aspect of the process with this implementation.